src/NServiceBus.Core/Hosting/StartupDiagnostics/HostStartupDiagnostics.cs

```diff
@@ -45,12 +45,23 @@ static WriteStartupDiagnostics GetDiagnosticsWriter(ReadOnlySettings settings)
                 catch (Exception e)
                 {
                     logger.Error("Unable to determine the diagnostics output directory. Check the attached exception for further information, or configure a custom diagnostics directory using 'EndpointConfiguration.SetDiagnosticsPath()'.", e);
+
+                    return data => TaskEx.CompletedTask;
                 }
             }
 
             if (!Directory.Exists(diagnosticsRootPath))
             {
-                Directory.CreateDirectory(diagnosticsRootPath);
+                try
+                {
+                    Directory.CreateDirectory(diagnosticsRootPath);
+                }
+                catch(Exception e)
+                {
+                    logger.Error("Unable to create the diagnostics output directory. Check the attached exception for further information, or configure a custom diagnostics directory using 'EndpointConfiguration.SetDiagnosticsPath()'.", e);
```

Comment:
Given that this message is the same as above would it make sense to consolidate the strings or better to leave as is given there is not much redundancy?

Reply:
It's not the same message.

Reply:
ah yeah one word is different 🗡

Reply:
Yeah, I wanted to distinguish the cases more, and my original comment above was about asking if we should make them even more different than they already are.